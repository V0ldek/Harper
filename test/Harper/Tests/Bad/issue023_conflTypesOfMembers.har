value T a = {
    variant A = {
        foo :: a -> a;
        foo x = x;
    };

    variant B = {
        foo :: a -> Integer;
        foo x = 42;
    };
};