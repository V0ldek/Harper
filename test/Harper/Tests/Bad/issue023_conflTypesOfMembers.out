Error: cannot unify conflicting types: `(T a) -> (a -> Integer)`, `(T a) -> (a -> a)` resulting from declarations of the member `foo` of the type `T`.
Conflicting declarations:
 foo :: T a -> a -> a
Located at line 3 column 9
 foo :: T a -> a -> Integer
Located at line 8 column 9

Execution terminated with an error: type error.